### PR TITLE
Test parts of API with libFuzzer.

### PR DESCRIPTION
**Pull request by PatKamin**
Add a fuzz test generating API calls. Related to #566 .

### CODE REVIEW

File: test/conformance/fuzz/urConformanceFuzz.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                std::uniform_int_distribution<unsigned> distrib(
+                    0, platforms.size() - 1);
+                platform = platforms[distrib(generator)];
+            }
+            urDeviceGet(platform, device_type, num_entries, devices.data(),
+                        &num_devices);
+            devices.resize(num_devices);
+            break;
+        case UR_CONTEXT_CREATE:
+            if (!devices.empty()) {
```
Comment (pbalcer) [inline test/conformance/fuzz/urConformanceFuzz.cpp:60]: we should allow the devices array to be parametrized by the fuzzer.

**Review by pbalcer — COMMENTED** [review]
This is a better approach, but we should expand the input grammar to cover more cases for these APIs.

File: test/conformance/fuzz/urConformanceFuzz.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    std::vector<ur_device_handle_t> devices;
+    std::map<ur_usm_pool_handle_t, std::vector<void *>> pool_host_allocs;
+    std::vector<void *> no_pool_host_allocs;
+    ur_platform_handle_t platform = nullptr;
+    ur_context_handle_t context = nullptr;
+
+    uint32_t num_adapters;
+    uint32_t num_platforms;
+    uint32_t num_devices;
+} TestState;
```
Comment (pbalcer) [inline test/conformance/fuzz/urConformanceFuzz.cpp:38]: you never use this global.

**Review by pbalcer — APPROVED** [review]
This is a good start, but right now the only thing that is being fuzzed is the sequence of operations. We need to allow the fuzzer to also parametrize these operations.

File: test/fuzz/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+    set_tests_properties(${TEST_TARGET_NAME} PROPERTIES
+        LABELS "fuzz"
+        ENVIRONMENT
+            "XPTI_TRACE_ENABLE=1"
+            "XPTI_FRAMEWORK_DISPATCHER=$<TARGET_FILE:xptifw>"
+            "XPTI_SUBSCRIBERS=$<TARGET_FILE:collector>"
+            "UR_ADAPTERS_FORCE_LOAD=\"$<TARGET_FILE:ur_adapter_null>\""
+            "UR_ENABLE_LAYERS=UR_LAYER_TRACING")
+    target_compile_options(${TEST_TARGET_NAME} PRIVATE -g -fsanitize=fuzzer)
```
Comment (kswiecicki) [inline test/fuzz/CMakeLists.txt:27]: You use `-fsanitize=fuzzer` here and in `target_link_libraries`, is that intentional?

**Comment by PatKamin** [comment]
Replaced by #795 targeting adapters branch.

**Pull request closed without merging**